ts/ts-diabetes/src/test/java/se/inera/intyg/common/ts_diabetes/v4/model/converter/InternalToTransportTest.java

```diff
@@ -0,0 +1,130 @@
+import se.inera.intyg.common.support.model.common.internal.Relation;
+import se.inera.intyg.common.support.model.converter.util.ConverterException;
+import se.inera.intyg.common.support.services.BefattningService;
+import se.inera.intyg.common.support.stub.IntygTestDataBuilder;
+import se.inera.intyg.common.ts_diabetes.v4.model.internal.Allmant;
+import se.inera.intyg.common.ts_diabetes.v4.model.internal.Bedomning;
+import se.inera.intyg.common.ts_diabetes.v4.model.internal.Hypoglykemi;
+import se.inera.intyg.common.ts_diabetes.v4.model.internal.Ovrigt;
+import se.inera.intyg.common.ts_diabetes.v4.model.internal.TsDiabetesUtlatandeV4;
+import se.riv.clinicalprocess.healthcond.certificate.registerCertificate.v3.RegisterCertificateType;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {BefattningService.class})
+public class InternalToTransportTest {
+
+    private static URL getResource(String href) {
+        return Thread.currentThread().getContextClassLoader().getResource(href);
+    }
+
+    public static TsDiabetesUtlatandeV4 getUtlatande() {
+        return getUtlatande(null, null, null);
+    }
+
+    public static TsDiabetesUtlatandeV4 getUtlatande(RelationKod relationKod, String relationMeddelandeId, String referensId) {
+        TsDiabetesUtlatandeV4.Builder utlatande = TsDiabetesUtlatandeV4.builder();
+        utlatande.setId("1234567");
+        utlatande.setTextVersion("1.0");
+        GrundData grundData = IntygTestDataBuilder.getGrundData();
+
+        grundData.setSigneringsdatum(LocalDateTime.parse("2015-12-07T15:48:05"));
+
+        if (relationKod != null) {
+            Relation relation = new Relation();
+            relation.setRelationKod(relationKod);
+            relation.setMeddelandeId(relationMeddelandeId);
+            relation.setReferensId(referensId);
+            grundData.setRelation(relation);
+        }
+        utlatande.setGrundData(grundData);
+
+        utlatande.setAllmant(Allmant.builder().build());
+        utlatande.setBedomning(Bedomning.builder().build());
+        utlatande.setHypoglykemi(Hypoglykemi.builder().build());
+        utlatande.setOvrigt(Ovrigt.builder().build());
+
+        return utlatande.build();
+    }
+
+    @Test
+    public void testInternalToTransportConversion() throws Exception {
+        TsDiabetesUtlatandeV4 expected = getUtlatande();
+        RegisterCertificateType transport = se.inera.intyg.common.ts_diabetes.v4.model.converter.InternalToTransport.convert(expected);
+        TsDiabetesUtlatandeV4 actual = TransportToInternal.convert(transport.getIntyg());
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = ConverterException.class)
+    public void testInternalToTransportSourceNull() throws Exception {
+        se.inera.intyg.common.ts_diabetes.v4.model.converter.InternalToTransport.convert(null);
+    }
+
+    @Test
+    public void convertDecorateSvarPaTest() throws Exception {
```

Comment:
This comment applies to to the rest of the tests in this file. 

Are these tests relevant when 'Ärendekommunikation' isn't supportet?

Reply:
fixat